global-api/routes/city_locode_endpoint.py

```diff
@@ -0,0 +1,116 @@
+from main import app
+from fastapi import HTTPException
+from sqlalchemy import create_engine, text
+from sqlalchemy.orm import SessionLocal, sessionmaker
+from decouple import config
+import pandas as pd
+
+# Define the database engine
+engine = create_engine(config("DATABASE_URL"))
+
+# Define SessionLocal for database interactions
+SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+
+# Extract the data by locode, year and sector/subsector
+def db_query(locode, year, reference_number):
+    
+    with SessionLocal() as session:
+        query = text("SELECT * FROM Asset " +
```

Comment:
You might need to use quotes around the table name (Asset). Consider using lower case table names for ease of use (snake_case).